easypy/collections.py

```diff
@@ -553,27 +559,50 @@ def index(self, obj):
             if lookup_uid == uid:
                 return i
 
-    def get_next(self, obj):
-        uid = self._get_object_uid(obj)
-        if uid not in self._objects:
-            raise ObjectNotFound(self, (), dict(key=uid),)
-        next_uid = self._objects._OrderedDict__map[uid].next
-        try:
-            key = next_uid.key
-        except AttributeError:
-            key = next_uid.next.key
-        return self._objects[key]
-
-    def get_prev(self, obj):
-        uid = self._get_object_uid(obj)
-        if uid not in self._objects:
-            raise ObjectNotFound(self, (), dict(key=uid),)
-        prev_uid = self._objects._OrderedDict__map[uid].prev
-        try:
-            key = prev_uid.key
-        except AttributeError:
-            key = prev_uid.prev.key
-        return self._objects[key]
+    if SUPPORT_GET_SIBLING:
+        def get_next(self, obj):
+            uid = self._get_object_uid(obj)
+            if uid not in self._objects:
+                raise ObjectNotFound(self, (), dict(key=uid),)
+            next_uid = self._objects._OrderedDict__map[uid].next
+            try:
+                key = next_uid.key
+            except AttributeError:
+                key = next_uid.next.key
+            return self._objects[key]
+
+        def get_prev(self, obj):
+            uid = self._get_object_uid(obj)
+            if uid not in self._objects:
+                raise ObjectNotFound(self, (), dict(key=uid),)
+            prev_uid = self._objects._OrderedDict__map[uid].prev
+            try:
+                key = prev_uid.key
+            except AttributeError:
+                key = prev_uid.prev.key
+            return self._objects[key]
+    else:
+        def get_next(self, obj):
+            from warning import warn
+            warn("Couldn't import a pure-python OrderedDict so this will be a O(n) operation")
+
+            from itertools import cycle
+            l1, l2 = cycle(self._objects.values()), cycle(self._objects.values())
+            next(l2)
+            for a, b in zip(l1, l2):
+                if obj == a:
+                    return b
```

Comment:
why not `if obj==a: return next(l1)`  instead of using two iterators?

Reply:
The internal implementation is not part of this PR... especially when I'd prefer to remove this code completely.

To your question - It would work here, but not in the `get_prev`, where we need an object we've already passed in the iteration.

Reply:
Regarding `get_prev`, I'd rather keep the previous item each time instead of iterating twice, but if we are planning to remove this section then it doesn't matter.